envoy/stream_info/filter_state.h

```diff
@@ -87,52 +87,44 @@ class FilterState {
 
   /**
    * @param data_name the name of the data being looked up (mutable/readonly).
-   * @return a const reference to the stored data.
-   * An exception will be thrown if the data does not exist. This function
-   * will fail if the data stored under |data_name| cannot be dynamically
-   * cast to the type specified.
+   * @return a typed pointer to the stored data or nullptr if the data does not exist or the data
+   * type does not match the expected type.
    */
-  template <typename T> const T& getDataReadOnly(absl::string_view data_name) const {
-    const T* result = dynamic_cast<const T*>(getDataReadOnlyGeneric(data_name));
-    if (!result) {
-      ExceptionUtil::throwEnvoyException(
-          fmt::format("Data stored under {} cannot be coerced to specified type", data_name));
-    }
-    return *result;
+  template <typename T> const T* getDataReadOnly(absl::string_view data_name) const {
+    const auto* generic_result = getDataReadOnlyGeneric(data_name);
+    return generic_result == nullptr ? nullptr : dynamic_cast<const T*>(generic_result);
```

Comment:
nit I think you can remove the conditional here and collapse this into a single line. dynamic_cast on nullptr is safe so you can just return that directly. Same below.

Reply:
get.